File: android/app/src/main/java/com/momentum/app/models/MLPrediction.kt
package com.momentum.app.models

import java.time.LocalDate
import java.util.UUID

data class Prediction(
    val date: LocalDate,
    val predictedEnergy: EnergyLevel,
    val predictedFocus: Int,
    val predictedStress: StressLevel,
    val confidence: Double,
    val keyDriver: String
)

data class Pattern(
    val id: String = UUID.randomUUID().toString(),
    val type: PatternType,
    val description: String,
    val frequency: Double,
    val strength: Double,
    val actionable: Boolean,
    val suggestedAction: MicroAction? = null
)

enum class PatternType(val label: String) {
    Weekly("Weekly"),
    Daily("Daily"),
    Trigger("Trigger"),
    Correlation("Correlation"),
    Anomaly("Anomaly")
}

data class UserProfile(
    var baselineSleep: Double = 7.0,
    var baselineSteps: Int = 5000,
    var baselineHRV: Double = 40.0,
    var effectiveActions: MutableMap<String, Double> = mutableMapOf(),
    var triggerPatterns: MutableMap<String, Double> = mutableMapOf(),
    var daysTracked: Int = 0
)

data class FeatureVector(
    val dayOfWeek: Int,
    val hour: Int,
    val sleepHours: Double,
    val sleepQualityScore: Int,
    val steps: Int,
    val heartRateAvg: Double,
    val heartRateVariability: Double,
    val screenTimeMinutes: Double,
    val calendarIntensity: Int,
    val previousDayEnergy: Int,
    val previousDayStress: Int,
    val previousDayFocus: Int,
    val weekToDateSleepAvg: Double,
    val weekToDateStepsAvg: Int,
    val monthToDateFocusAvg: Int
)

File: ios/Momentum/Models/MLPrediction.swift
import Foundation

struct Prediction: Codable {
    let date: Date; let predictedEnergy: EnergyLevel; let predictedFocus: Int; let predictedStress: StressLevel; let confidence: Double; let keyDriver: String
}

struct Pattern: Codable, Identifiable {
    let id: UUID; let type: PatternType; let description: String; let frequency: Double; let strength: Double; let actionable: Bool; let suggestedAction: MicroAction?
    enum PatternType: String, Codable { case weekly = "Weekly"; case daily = "Daily"; case trigger = "Trigger"; case correlation = "Correlation"; case anomaly = "Anomaly" }
}

struct UserProfile: Codable {
    var baselineSleep: Double = 7.0; var baselineSteps: Int = 5000; var baselineHRV: Double = 40.0; var effectiveActions: [String: Double] = [:]; var triggerPatterns: [String: Double] = [:]; var weeklyPattern: [Int: Double] = [:]; var learningVersion: Int = 1; var daysTracked: Int = 0
}

struct FeatureVector: Codable {
    let dayOfWeek: Int; let hour: Int; let sleepHours: Double; let sleepQualityScore: Int; let steps: Int; let heartRateAvg: Double; let heartRateVariability: Double; let screenTimeMinutes: Double; let calendarIntensity: Int; let previousDayEnergy: Int; let previousDayStress: Int; let previousDayFocus: Int; let weekToDateSleepAvg: Double; let weekToDateStepsAvg: Int; let monthToDateFocusAvg: Int
}
